src/recipes.jl

```diff
@@ -635,7 +635,15 @@ _hist_edges(vs::NTuple{N,AbstractVector}, binning::Union{Integer, Symbol, Abstra
 _hist_norm_mode(mode::Symbol) = mode
 _hist_norm_mode(mode::Bool) = mode ? :pdf : :none
 
+_filternans(vs::NTuple{1,AbstractVector}) = filter!.(!isnan, vs)
+function _filternans(vs::NTuple{N,AbstractVector}) where N
+    _invertedindex(v, not) = [j for (i,j) in enumerate(v) if !(i ∈ not)]
+    nots = union(Set.(findall.(isnan, vs))...)
+    _invertedindex.(vs, Ref(nots))
+end
+
 function _make_hist(vs::NTuple{N,AbstractVector}, binning; normed = false, weights = nothing) where N
+    vs = _filternans(vs)
```

Comment:
Re-binding `vs` may lead to trouble ... probably better to give either the filtered data or the function argument a different name.

Reply:
It'll be hard to guarantee that the type won't change, right? In any case, I think it would be cleaner. And I vaguely remember I read somewhere that re-binding function arguments can lead to problems, but I have to admit I forgot when and why.

Reply:
OK I'll do it and merge, doesn't really matter. But I'd think my _filternan function is guaranteed to return a Tuple{Vector{Float64}}. Thanks for taking a look!